#4083 fixed external link warning when user wants to make a donation

viditpawar0 opened the pull request:
<!--
- Fixes #

- If you haven't solved the issue completely use "Linked issue #{issue_number}.
- After solving the issue completely change it to "Fixes #{issue_number}.
-->
Fixes #4083 

<!-- Add here what changes were made in this issue and if possible provide links. -->
Fixed external link warning when user wants to make a donation

<!-- If possible, please add relevant screenshots / GIFs -->

MohitMaliFtechiz commented on core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt:
```diff
@@ -280,7 +280,7 @@ abstract class CoreMainActivity : BaseActivity(), WebViewProvider {
 
   fun openSupportKiwixExternalLink() {
-    externalLinkOpener.openExternalUrl(KIWIX_SUPPORT_URL.toUri().browserIntent())
+    externalLinkOpener.openExternalUrl(KIWIX_SUPPORT_URL.toUri().browserIntent(), false)
```
@viditpawar0 We have the same functionality for custom apps as well. So in `CustomMainActivity` we have a function `openExternalUrl`. With this function, we are opening the `support_url` which is the same as the Kiwix app so we should refactor the code there as well.

MohitMaliFtechiz reviewed (changes requested):
@viditpawar0 We have the `DonationPopup` functionality as well which we have recently added, which shows on the reader screen. So when the user clicks on the "Make a donation" button this method calls https://github.com/kiwix/kiwix-android/blob/9fc6e66ecc00ddcc6b93e958934ff3f2ccbbaa8b/core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt#L1676

According to your existing code, it will work fine with the Kiwix app, but for custom apps, we have overrided this method in  `CustomReaderFragment` to make custom behavior according to configuration. So we have to make the changes there as well so that when while user clicks on the donation button in custom apps this popup will not show.


Apart from these changes: The test cases are falling due to this change specially `topLevelDestinationTest` so can you please fix that?

viditpawar0 commented:
Sure, I'll do that

viditpawar0 commented:
Done

viditpawar0 commented:
All the requested changes have been done.

Summary: -
- made the same fix for custom apps in `CustomMainActivity`
- modified `topLevelDestinationTest`
- fixed the same for `CoreReaderFragment` and `CustomReaderFragment`

MohitMaliFtechiz reviewed (changes requested):
@viditpawar0 Thanks for making the changes. Please see comments.

viditpawar0 commented:
Okay, made the required changes

MohitMaliFtechiz reviewed (approved):
@viditpawar0 Thanks for making the changes. @kelson42 LGTM.
